lib/DxilValidation/DxilContainerValidation.cpp

```diff
@@ -1033,8 +1033,28 @@ HRESULT ValidateDxilContainerParts(llvm::Module *pModule,
     case DFCC_ResourceDef:
     case DFCC_ShaderStatistics:
     case DFCC_PrivateData:
+      break;
     case DFCC_DXIL:
-    case DFCC_ShaderDebugInfoDXIL:
+    case DFCC_ShaderDebugInfoDXIL: {
+      const DxilProgramHeader *pProgramHeader =
+          reinterpret_cast<const DxilProgramHeader *>(GetDxilPartData(pPart));
+      if (pProgramHeader) {
```

Comment:
nit: you could continue on `!pProgramHeader` to reduce nesting.
https://llvm.org/docs/CodingStandards.html#use-early-exits-and-continue-to-simplify-code